### ISSUE TITLE
2.0.2.9 crashes with System.IO.FileNotFoundException, missing file or assembly "CoCoL"

### ISSUE THREAD

**Issue by milux** [duplicati/duplicati#2788]
- [ x ] I have searched open and closed issues for duplicates.

----------------------------------------

## Environment info

- **Duplicati version**: 2.0.2.9
- **Operating system**: Windows 10 x64
- **Backend**: MEGA

## Description

When trying to start Duplicati.GUI.TrayIcon.exe, it immediately crashes. When starting in cmd.exe (Admin
privileges), it throws the following error message / stack traces: (Error text is given in German because of the operating system language):
```
Crash!
System.Reflection.TargetInvocationException: Ein Aufrufziel hat einen Ausnahmefehler verursacht. ---> System.IO.FileNotFoundException: Die Datei oder Assembly "CoCoL, Version=1.5.0.24305, Culture=neutral, PublicKeyToken=0983de3c914beeaa" oder eine Abhängigkeit davon wurde nicht gefunden. Das System kann die angegebene Datei nicht finden.
   bei Duplicati.Library.UsageReporter.Reporter.ShutDown()
   bei Duplicati.Server.Program.RealMain(String[] args)
   --- Ende der internen Ausnahmestapelüberwachung ---
   bei System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   bei System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   bei System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   bei Duplicati.Library.AutoUpdater.UpdaterManager.RunMethod(MethodInfo method, String[] args)
Crash!
System.Reflection.TargetInvocationException: Ein Aufrufziel hat einen Ausnahmefehler verursacht. ---> System.IO.FileNotFoundException: Die Datei oder Assembly "CoCoL, Version=1.5.0.24305, Culture=neutral, PublicKeyToken=0983de3c914beeaa" oder eine Abhängigkeit davon wurde nicht gefunden. Das System kann die angegebene Datei nicht finden.
   bei Duplicati.GUI.TrayIcon.HostedInstanceKeeper..ctor(String[] args)
   bei Duplicati.GUI.TrayIcon.Program.RealMain(String[] _args)
   --- Ende der internen Ausnahmestapelüberwachung ---
   bei System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   bei System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   bei System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   bei Duplicati.Library.AutoUpdater.UpdaterManager.RunMethod(MethodInfo method, String[] args)

Unbehandelte Ausnahme: System.IO.FileNotFoundException: Die Datei oder Assembly "CoCoL, Version=1.5.0.24305, Culture=neutral, PublicKeyToken=0983de3c914beeaa" oder eine Abhängigkeit davon wurde nicht gefunden. Das System kann die angegebene Datei nicht finden.
   bei Duplicati.Library.AutoUpdater.UpdaterManager.RunMethod(MethodInfo method, String[] args)
   bei Duplicati.Library.AutoUpdater.UpdaterManager.<>c__DisplayClass53_0.<RunFromMostRecent>b__0()
   bei Duplicati.Library.AutoUpdater.UpdaterManager.WrapWithUpdater(AutoUpdateStrategy defaultstrategy, Action wrappedFunction)
   bei Duplicati.Library.AutoUpdater.UpdaterManager.RunFromMostRecent(MethodInfo method, String[] cmdargs, AutoUpdateStrategy defaultstrategy)
   bei Duplicati.GUI.TrayIcon.Program.Main(String[] args)
   bei System.AppDomain._nExecuteAssembly(RuntimeAssembly assembly, String[] args)
   bei System.AppDomain.ExecuteAssemblyByName(String assemblyName, String[] args)
   bei System.AppDomain.ExecuteAssemblyByName(String assemblyName, String[] args)
   bei Duplicati.Library.AutoUpdater.UpdaterManager.RunFromMostRecent(MethodInfo method, String[] cmdargs, AutoUpdateStrategy defaultstrategy)
   bei Duplicati.GUI.TrayIcon.Program.Main(String[] args)
```

## Steps to reproduce

1. Updated from 2.0.2.8 to 2.0.2.9 (Direct update via MSI, **not** using internal updater!)
2. Tried to start Duplicati Tray executable as usual

**Comment by vb0** [comment]
I'm getting the same. I updated over the internal updater.
What is more disturbing (and it was the same every time something breaks completely) is that installing older version(s) that were working doesn't change anything.

**Comment by jidea** [comment]
Same problem but downgrade to 2.0.2.8 was fine. I wait for an another update.

**Comment by milux** [comment]
@vb0 Installing an old version and performing a "repair" operation fixed your issue for me. It seems that some executables are missing after the downgrade.

**Comment by vb0** [comment]
It's working now, I had to remove C:\Users\All Users\Duplicati\updates\2.0.2.9 to make the server work too.

**Comment by travisdietrich** [comment]
Same issue, downgrading didn't help.  vb0's suggestion to delete the folder above seems to have resolved, 2.0.2.8 is launching again.

**Comment by travisdietrich** [comment]
Tried the new 2.0.2.10 and same issue.  Rolled back to 2.0.2.8 again.

**Comment by milux** [comment]
Read the changelog and decided that it wasn't worth to try... seems that I was right. ^^

**Comment by kenkendk** [comment]
Related forum post:
https://forum.duplicati.com/t/duplicati-2-0-2-9-won-t-start/885/22?u=kenkendk

**Comment by darksworm** [comment]
I've encountered this issue and found the problem.
The problem was that in the newer installer, the version of CoCol.dll was lower than the one in the already installed version of duplicati and there is a bug in WiX - when trying to install a lower version dll, the old one will get removed but because the dll in the new installer is older it won't be installed. 

tl;dr - downgrading DLL files does not work, the file will be removed but not replaced.

I've hacked my way through the problem by increasing the version of CoCol.dll with resource hacker.
